src/components/NotificationsDrawer/DrawerPanelContent.tsx

```diff
@@ -109,14 +119,53 @@ const DrawerPanelBase = ({ innerRef }: DrawerPanelProps) => {
     dispatch(toggleNotificationsDrawer());
   };
 
-  const onMarkAllAsRead = () => {
-    dispatch(markAllNotificationsAsRead());
+  const onMarkSelectedAsRead = () => {
     setIsDropdownOpen(false);
+    dispatch(markSelectedNotificationsAsRead());
+
+    axios
+      .put('/api/notifications/v1/notifications/drawer/read', {
+        notification_ids: notifications.filter(({ selected }) => selected).map(({ id }) => id),
+        read_status: true,
+      })
+      .then(() => {
+        dispatch(deselectAllNotifications());
+      })
+      .catch((error) => {
+        dispatch(markSelectedNotificationsAsUnread());
+        console.error('Failed to update notification read status', error);
+      });
   };
 
-  const onMarkAllAsUnread = () => {
-    dispatch(markAllNotificationsAsUnread());
+  const onMarkSelectedAsUnread = () => {
     setIsDropdownOpen(false);
+    dispatch(markSelectedNotificationsAsUnread());
+
+    axios
+      .put('/api/notifications/v1/notifications/drawer/read', {
+        notification_ids: notifications.filter(({ selected }) => selected).map(({ id }) => id),
+        read_status: false,
+      })
+      .then(() => {
+        dispatch(deselectAllNotifications());
+      })
+      .catch((error) => {
+        dispatch(markSelectedNotificationsAsRead());
+        console.error('Failed to update notification read status', error);
+      });
+  };
```

Comment:
It's not ideal to dispatch action and then run async function. Ideally we can have action in `/src/redux/actions` that dispatches async function. Once it gets back to us we can update `/src/redux/index` to include new reducer and update the selected events in there.